### ISSUE TITLE
[BUG] Accessing alt on ExternalImage throws error

### ISSUE THREAD

**Issue by Paul-Hebert** [timber/timber#2817]
### Expected Behavior

The following code should not throw an error and crash the site:

```php
$url = 'https://paulmh.xyz/art/get-out-of-your-head/get-out-of-your-head.jpg';
$context['cover_image'] = Timber::get_external_image($url);
```

```twig
<img src="{{ cover_image.src }}" alt="{{cover_image.alt}}" />
```

(I'm a little confused about how you would set the `alt` on an ExternalImage. My real-world use case is a twig template that accepts a Timber Image and can be used for external or local images. For external images I would expect to get an empty string back.)

### Actual behavior

I get an error, and my site crashes

```
Fatal  error: Uncaught Error: Typed property Timber\ExternalImage::$alt_text  must not be accessed before initialization in  /Users/paulhebert/repos/cloudfour/app/public/vendor/timber/timber/src/ExternalImage.php  on line 476
```

<img width="961" alt="image" src="https://github.com/timber/timber/assets/5798536/976b7d65-a675-4ef0-8fcd-ffa19a06936a">

### Steps to reproduce behavior

1. Add an external image to your page context using `Timber::get_external_image`
2. In your twig template, access the alt property on the image.

### What version of Timber are you using?

v2.0.0 RC.1

### What version of WordPress are you using?

6.3.2

### What version of PHP are you using?

8.2.10

### How did you install Timber?

Installed or updated Timber through Composer

**Comment by Paul-Hebert** [comment]
Digging in the source a bit, I think the bug lies in how `ExternalImage:build` works. On line 150 there's the following code:

```php
        if (!empty($args['alt'])) {
            $external_image->alt_text = (string) $args['alt'];
        }
```

This means if there's no alt, or alt is an empty string then the property is never initialized and it throws an error when you access it. I'd prefer the value to be set to an empty string in this scenario so that if I try to access it I don't get errors.

Something like this:

```php
        if (!empty($args['alt'])) {
            $external_image->alt_text = (string) $args['alt'];
        } else {
            $external_image->alt_text = '';
        }
```

Would you be open to a PR with this change? Alternately, is there a way within my Twig template to check if the variable has been initialized or not?

**Comment by jrathert** [comment]
> Alternately, is there a way within my Twig template to check if the variable has been initialized or not?

Yes, there is, the `default` filter: `{{ alt|default('') }}` 

See https://twig.symfony.com/doc/3.x/filters/default.html

**Comment by Paul-Hebert** [comment]
Thanks @jrathert ! Unfortunately both of those options still throw the same error 🤔

**Comment by jrathert** [comment]
Ooops. Yes, you are right. It seems this bug is actually caused by the fact, that the respective property of `ExternalImage`:
```
    protected string $alt_text;
```
is not initialized as it is a typed, non-nullable property and you are using PHP >=7.4. (This behavior - not initializing typed properties - was only introduced in PHP 7.4). We have the same problem with the `caption` property.

Your workaround is probably a good solution for now, but of course fixing the root cause (probably by initializing `$alt_text` and `$caption` to `""`, or by always setting the respective values in the `build()` function (see lines 150-156)) would be the better solution.

**Comment by Paul-Hebert** [comment]
That makes sense

>Your workaround is probably a good solution for now, but of course fixing the root cause (probably by initializing $alt_text and $caption to "", or by always setting the respective values in the build() function (see lines 150-156)) would be the better solution.

Yeah, this makes sense to me. Even initializing it to `null` might work while being more explicit?

**Comment by jrathert** [comment]
Initializing to `null` will only work if we change the type to `?string`, and also the signature of the respective methods. Do not know what side-effect that would have.

The whole code is a bit, let's say, interesting: The `$args` value gets set to default values (`alt` and `caption` in `Timber::get_external_image()`, but only `alt` in `ExternalImage::build()` - but these default values have no effect anyway as they are considered empty in lines 150-156 of `ExternalImage.php`. Why setting them at all, then? (And btw. removing the test for emptiness would also be a solution to the problem...)

**Comment by jrathert** [comment]
@Paul-Hebert - I submitted a PR fixing this issue. I did not want to make it too difficult, so decided to use empty strings as default values for both, `alt` and `caption`. Let's see what the team says. Thanks for raising it!

**Comment by Paul-Hebert** [comment]
Awesome, thanks @jrathert!

**Comment by gchtr** [comment]
Resolved in #2818.
